File: typescript/base-hmi-types/src/screens/controls/HmiAlarmEventSubscription.ts
import { HmiProperty } from "../base.js";
import { HmiAlarmEventPriority } from "./HmiAlarmEventPriority.js";

export class HmiAlarmEventSubscription {
  name?: string;
  isDefault?: HmiProperty<boolean>;
  readonly priorities: HmiAlarmEventPriority[] = [];
  readonly sourcePriorities: string[] = [];
  readonly scopes: string[] = [];
  readonly eventSources: string[] = [];
}

File: src/BaseHmiTypes/Screens/Controls/HmiAlarmEventSubscription.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAlarmEventSubscription
{
    public string? Name { get; set; }

    public HmiProperty<bool>? IsDefault { get; set; }

    public IList<HmiAlarmEventPriority> Priorities { get; } = new List<HmiAlarmEventPriority>();

    public IList<string> SourcePriorities { get; } = new List<string>();

    public IList<string> Scopes { get; } = new List<string>();

    public IList<string> EventSources { get; } = new List<string>();
}
